src/main/java/org/testmonkeys/jentitytest/comparison/property/StringComparator.java

```diff
@@ -0,0 +1,52 @@
+package org.testmonkeys.jentitytest.comparison.property;
+
+import org.testmonkeys.jentitytest.comparison.ComparisonContext;
+import org.testmonkeys.jentitytest.comparison.SingleResultComparator;
+import org.testmonkeys.jentitytest.comparison.conditionalChecks.NullConditionalCheck;
+import org.testmonkeys.jentitytest.comparison.result.ComparisonResult;
+import org.testmonkeys.jentitytest.framework.JEntityTestException;
+
+public class StringComparator extends SingleResultComparator {
+    private boolean caseSensitive = true;
+    private boolean ignoreTrailing = false;
+    private char[] ignoreCharacters = {};
+
+    public StringComparator() {
+        registerPreConditionalCheck(new NullConditionalCheck());
+    }
+
+    public ComparisonResult computeSingleComparison(Object actualValue, Object expectedValue, ComparisonContext context) {
+        ComparisonResult comparisonResult = new ComparisonResult(false, context, actualValue, expectedValue);
+
+        String actual;
+        String expected;
+
+        try {
+            actual = (String) actualValue;
+            expected = (String) expectedValue;
+        } catch (ClassCastException castException) {
+            throw new JEntityTestException("Expected and Actual values must of type " + String.class.getName(),
+                    castException);
+        }
+
+        if (actual != null && expected != null) {
+            for (char ignore : ignoreCharacters) {
+
+                String ignoreString = String.valueOf(ignore);
```

Comment:
we should be using the value of ignore directly in replace, no need to transform to String